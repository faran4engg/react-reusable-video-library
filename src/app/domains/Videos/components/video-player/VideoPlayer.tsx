import { FC } from 'react';
import ReactPlayer from 'react-player/lazy';
import { VideoPlayerProps } from './types';
import '../styles.css';

const VideoPlayer: FC<VideoPlayerProps> = ({
  url,
  watchVideo,
  disbaleVideoClick = false,
  isLightMode = true,
}) => (
  <div className="aspect-ratio-box" onClick={watchVideo}>
    {disbaleVideoClick && (
      <div className="absolute top-0 left-0 z-10 w-full h-full cursor-pointer"></div>
    )}
    <ReactPlayer
      className="react-player"
      playIcon={<span>&nbsp;</span>}
      light={isLightMode}
      controls
      width="100%"
      height="100%"
      url={url}
      config={{
        youtube: {
          playerVars: { modestbranding: 1, rel: 0 },
        },
        dailymotion: {
          params: {
            'sharing-enable': false,
            'ui-logo': false,
            'ui-start-screen-info': false,
            'queue-enable': false,
          },
        },
      }}
    />
  </div>
);

export default VideoPlayer;
