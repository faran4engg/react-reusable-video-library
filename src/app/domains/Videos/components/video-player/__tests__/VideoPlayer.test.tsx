import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import VideoPlayer from '../VideoPlayer';

const renderer = createRenderer();

describe('<VideoPlayer />', () => {
  it('should render and VideoPlayer should not be clickable', () => {
    renderer.render(<VideoPlayer url="yt.com/asd" disbaleVideoClick />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
  it('should render and VideoPlayer should be clickable', () => {
    renderer.render(
      <VideoPlayer
        url="yt.com/asd"
        disbaleVideoClick={false}
        isLightMode={false}
      />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
