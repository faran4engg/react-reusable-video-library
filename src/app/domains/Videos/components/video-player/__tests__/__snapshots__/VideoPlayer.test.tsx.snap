// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlayer /> should render and VideoPlayer should be clickable 1`] = `
<div
  className="aspect-ratio-box"
>
  <ReactPlayer
    className="react-player"
    config={
      Object {
        "dailymotion": Object {
          "params": Object {
            "queue-enable": false,
            "sharing-enable": false,
            "ui-logo": false,
            "ui-start-screen-info": false,
          },
        },
        "youtube": Object {
          "playerVars": Object {
            "modestbranding": 1,
            "rel": 0,
          },
        },
      }
    }
    controls={true}
    height="100%"
    light={false}
    loop={false}
    muted={false}
    onBuffer={[Function]}
    onBufferEnd={[Function]}
    onDisablePIP={[Function]}
    onDuration={[Function]}
    onEnablePIP={[Function]}
    onEnded={[Function]}
    onError={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onProgress={[Function]}
    onReady={[Function]}
    onSeek={[Function]}
    onStart={[Function]}
    pip={false}
    playIcon={
      <span>
         
      </span>
    }
    playbackRate={1}
    playing={false}
    playsinline={false}
    progressInterval={1000}
    stopOnUnmount={true}
    style={Object {}}
    url="yt.com/asd"
    volume={null}
    width="100%"
    wrapper="div"
  />
</div>
`;

exports[`<VideoPlayer /> should render and VideoPlayer should not be clickable 1`] = `
<div
  className="aspect-ratio-box"
>
  <div
    className="absolute top-0 left-0 z-10 w-full h-full cursor-pointer"
  />
  <ReactPlayer
    className="react-player"
    config={
      Object {
        "dailymotion": Object {
          "params": Object {
            "queue-enable": false,
            "sharing-enable": false,
            "ui-logo": false,
            "ui-start-screen-info": false,
          },
        },
        "youtube": Object {
          "playerVars": Object {
            "modestbranding": 1,
            "rel": 0,
          },
        },
      }
    }
    controls={true}
    height="100%"
    light={true}
    loop={false}
    muted={false}
    onBuffer={[Function]}
    onBufferEnd={[Function]}
    onDisablePIP={[Function]}
    onDuration={[Function]}
    onEnablePIP={[Function]}
    onEnded={[Function]}
    onError={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onProgress={[Function]}
    onReady={[Function]}
    onSeek={[Function]}
    onStart={[Function]}
    pip={false}
    playIcon={
      <span>
         
      </span>
    }
    playbackRate={1}
    playing={false}
    playsinline={false}
    progressInterval={1000}
    stopOnUnmount={true}
    style={Object {}}
    url="yt.com/asd"
    volume={null}
    width="100%"
    wrapper="div"
  />
</div>
`;
